### PR TITLE
Unify all hibernate config files and change session context to thread

### PR DESCRIPTION
**Pull request by Beacze**
current_session_context_class: http://docs.jboss.org/hibernate/orm/4.3/manual/en-US/html/ch02.html#architecture-current-session

### CODE REVIEW

File: Kitodo/src/test/java/org/kitodo/MockDatabase.java [hunk 1]
```diff
@@ -824,10 +824,10 @@ private static void insertProjectFileGroups() throws DAOException, DataException
         fifthProjectFileGroup.setProject(project);
         serviceManager.getProjectFileGroupService().saveToDatabase(fifthProjectFileGroup);
 
-        project.getProjectFileGroups().add(fifthProjectFileGroup);
-        project.getProjectFileGroups().add(fifthProjectFileGroup);
-        project.getProjectFileGroups().add(fifthProjectFileGroup);
-        project.getProjectFileGroups().add(fifthProjectFileGroup);
+        project.getProjectFileGroups().add(firstProjectFileGroup);
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/MockDatabase.java]: This changes did not fit in this commit nor pull request. Can you please move it to a separate commit or better pull request?

File: Kitodo/src/main/resources/hibernate.cfg.xml [hunk 2]
```diff
@@ -52,7 +52,7 @@
         <property name="hibernate.cache.region.factory_class">org.hibernate.cache.ehcache.EhCacheRegionFactory</property>
 
         <!-- Enable Hibernate's automatic session context management -->
-        <property name="current_session_context_class">managed</property>
+        <property name="current_session_context_class">thread</property>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/resources/hibernate.cfg.xml]: What is the reason to switch from `managed` to `thread`? What kind of issue did you fix with this change?

Comment (Beacze) [reply Kitodo/src/main/resources/hibernate.cfg.xml]: thread: current sessions are tracked by thread of execution.

managed: current sessions are tracked by thread of execution. However, you are responsible to bind and unbind a Session instance with static methods on this class: it does not open, flush, or close a Session.

I suspect that thread is more forgiving than managed...

Comment (henning-gerhardt) [reply Kitodo/src/main/resources/hibernate.cfg.xml]: Thank you for the explanation.
